src/ray/raylet_ipc_client/client_connection.cc

```diff
@@ -274,18 +274,20 @@ void ServerConnection::DoAsyncWrites() {
   if (RayConfig::instance().event_stats()) {
     auto &io_context =
         static_cast<instrumented_io_context &>(socket_.get_executor().context());
+    auto event_stats = io_context.stats();
     const auto stats_handle =
-        io_context.stats().RecordStart("ClientConnection.async_write.DoAsyncWrites");
+        event_stats->RecordStart("ClientConnection.async_write.DoAsyncWrites");
     boost::asio::async_write(
         socket_,
         message_buffers,
         [this,
          this_ptr,
          num_messages,
          call_handlers,
+         event_stats,
          stats_handle = std::move(stats_handle)](const boost::system::error_code &error,
                                                  size_t bytes_transferred) {
-          EventTracker::RecordExecution(
+          event_stats->RecordExecution(
```

Comment:
We don't need to capture `event_stats`. We have `this` so we can get `static_cast<instrumented_io_context &>(socket_.get_executor().context());` again to get the event_stats. Trying to minimizing the things we capture if possible.